app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -141,57 +138,61 @@ class InfectionMessengerRepositoryImpl(
-        if (lastScheduledToken != token){
+        if (lastScheduledToken != token) {
             Timber.e(IllegalArgumentException("processing of token ${token} when we expect to process ${lastScheduledToken}"))
         }
 
-        if (token.startsWith(BATCH)){
-            processKeysBasedOnBatchToken(token)
-        } else {
-            processDailyTokens(token)
-        }
+        processKeysBasedOnBatchToken(token)
     }
 
     private fun processDailyTokens(token: String) {
         //we can assume you are WarningType.GREEN as only in this case we process the tokens
     }
 
     private suspend fun processKeysBasedOnBatchToken(token: String) {
-        val warningType = WarningType.RED
+        val warningType = quarantineRepository.getCurrentWarningType()
 
         val summary = exposureNotificationRepository.determineRiskWithoutInformingUser(token)
         val configuration =
             configurationRepository.getConfiguration()
                 ?: throw IllegalStateException("we have no configuration values here, it doesn´t make sense to continue")
 
-        when(warningType) {
-
+        when (warningType) {
             WarningType.YELLOW, WarningType.RED -> {
                 //TODO finish this decision branch
-                if (summary.summationRiskScore >= configuration.dailyRiskThreshold){
+                if (summary.summationRiskScore >= configuration.dailyRiskThreshold) {
                     val summary = exposureNotificationRepository.getExposureSummaryWithPotentiallyInformingTheUser(token)
                     //go through the days and check if the day is the first RED/YELLOW day
                     // TODO: go through the summary and check if the day is the first RED/YELLOW day
                 } else {
-                    //TODO: we are green again
-                    //TODO finish this decision branch
+                    when (warningType) {
+                        WarningType.RED -> quarantineRepository.revokeLastRedContactDate()
+                        WarningType.YELLOW -> quarantineRepository.revokeLastYellowContactDate()
+                    }
+                    quarantineRepository.setShowQuarantineEnd()
                 }
             }
             WarningType.REVOKE -> {
                 //we are above risc for the last days!!!
-                if (summary.summationRiskScore >= configuration.dailyRiskThreshold){
+                if (summary.summationRiskScore >= configuration.dailyRiskThreshold) {
                     //we must now identify day by day if we are YELLOW or RED
-                    val listOfDaysWithDownloadedFilesSortedByServer = apiInteractor.fetchDailyBatchDiagnosisKeys()
+                    //val listOfDaysWithDownloadedFilesSortedByServer = apiInteractor.fetchDailyBatchDiagnosisKeys()
+                    //process the batches and do one of these:
+                    //TODO find time of contact
+                    //this is a fake calculation:
```

Comment:
not sure if this should go into production?

Reply:
it will only go to the story branch...